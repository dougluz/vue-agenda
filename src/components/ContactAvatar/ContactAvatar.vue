<script setup lang="ts">
import { computed } from "vue";

type ContactAvatarProps = {
  contactName: string | undefined;
};

const props = defineProps<ContactAvatarProps>();

const getContactFirstLetter = computed(() => {
  if (props.contactName) {
    return props.contactName[0].toUpperCase();
  }
  return "?";
});
</script>

<template>
  <span class="avatar">{{ getContactFirstLetter }}</span>
</template>

<style scoped>
.avatar {
  @apply w-6 h-6 bg-salmon text-white rounded-full;
  padding: 0.188rem 0.313rem 0.125rem 0.375rem;
}
</style>
