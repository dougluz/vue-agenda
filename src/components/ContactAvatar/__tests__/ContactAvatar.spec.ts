import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import ContactAvatar from "../ContactAvatar.vue";

describe("ContactAvatar", () => {
  describe("when receiving a contact with a name", () => {
    it("should match the snapshot", () => {
      const wrapper = mount(ContactAvatar, {
        props: {
          contactName: "John Doe",
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
    it("should show the first letter of the name", () => {
      const wrapper = mount(ContactAvatar, {
        props: {
          contactName: "John Doe",
        },
      });
      expect(wrapper.text()).toBe("J");
    });
  });
  describe("when receiving a contact without a name", () => {
    it("should match the snapshot", () => {
      const wrapper = mount(ContactAvatar);
      expect(wrapper.html()).toMatchSnapshot();
    });

    it("should show the ? as replacement to the first letter", () => {
      const wrapper = mount(ContactAvatar);
      expect(wrapper.text()).toBe("?");
    });
  });
});
