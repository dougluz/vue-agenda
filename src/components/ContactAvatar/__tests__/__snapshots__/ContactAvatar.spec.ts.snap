// Vitest Snapshot v1

exports[`ContactAvatar > when receiving a contact with a name > should match the snapshot 1`] = `"<span class=\\"avatar\\" data-v-689e67aa=\\"\\">J</span>"`;

exports[`ContactAvatar > when receiving a contact without a name > should match the snapshot 1`] = `"<span class=\\"avatar\\" data-v-689e67aa=\\"\\">?</span>"`;
